tests/interfaces/test_autograd.py

```diff
@@ -475,3 +475,234 @@ def d_classical(a, b, method):
+
+        @qml.qnode(dev, interface="autograd")
+        def circuit(data1, weights, data2):
+            # non-differentiable quantum function
+            qml.templates.AmplitudeEmbedding(data1, wires=[0, 1])
+            # differentiable quantum function
+            qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1])
+            # non-differentiable quantum function
+            qml.templates.AngleEmbedding(data2, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        # differentiating the circuit wrt the weights
+        grad_fn = qml.grad(circuit)
+
+        # input weights
+        weights = qml.init.strong_ent_layers_normal(n_wires=2, n_layers=3)
+
+        # input data
+        data1 = qml.numpy.array([0, 1, 1, 0], requires_grad=False) / np.sqrt(2)
+        data2 = qml.numpy.array([1, 1], requires_grad=False)
+
+        res = grad_fn(data1, weights, data2)
+
+        # we do not check for correctness, just that the output
+        # is the correct shape
+        assert len(res) == 1
+        assert res[0].shape == weights.shape
+
+        # check that the parameter shift was only performed for the
+        # differentiable input, not the data input
+        assert spy.call_args[1]["wrt"] == set(data1.size + np.arange(weights.size))
+
+    def test_differentiable_parameter_last(self, mocker):
+        """Test that a differentiable parameter used as the last
+        argument is correctly evaluated by QNode.jacobian, and that
+        all other non-differentiable parameters are ignored"""
+        spy = mocker.spy(qml.qnodes.JacobianQNode, "jacobian")
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface="autograd")
+        def circuit(data1, data2, weights):
+            # non-differentiable quantum function
+            qml.templates.AmplitudeEmbedding(data1, wires=[0, 1])
+            # differentiable quantum function
+            qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1])
+            # non-differentiable quantum function
+            qml.templates.AngleEmbedding(data2, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        # differentiating the circuit wrt the weights
+        grad_fn = qml.grad(circuit)
+
+        # input weights
+        weights = qml.init.strong_ent_layers_normal(n_wires=2, n_layers=3)
+
+        # input data
+        data1 = qml.numpy.array([0, 1, 1, 0], requires_grad=False) / np.sqrt(2)
+        data2 = qml.numpy.array([1, 1], requires_grad=False)
+
+        res = grad_fn(data1, data2, weights)
+
+        # we do not check for correctness, just that the output
+        # is the correct shape
+        assert len(res) == 1
+        assert res[0].shape == weights.shape
+
+        # check that the parameter shift was only performed for the
+        # differentiable input, not the data input
+        assert spy.call_args[1]["wrt"] == set(data1.size + data2.size + np.arange(weights.size))
```

Comment:
Same question; the `assert` doesn't seem to intuitively match my expectation from the comment